components/HTMLBlock/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+const MATCH_SCRIPT_TAGS = /<script\b[^>]*>([\s\S]*?)<\/script *>\n?/gim;
+
+const extractScripts = (html: string = ''): [string, () => void] => {
+  const scripts: string[] = [];
+  let match: RegExpExecArray | null;
+  while ((match = MATCH_SCRIPT_TAGS.exec(html)) !== null) {
+    scripts.push(match[1]);
+  }
+  const cleaned = html.replace(MATCH_SCRIPT_TAGS, '');
+  return [cleaned, () => scripts.map(js => window.eval(js))];
```

Comment:
Like, there's no reason to strip script tags if the whole thing is getting executed?

Reply:
it only gets executed if you pass `runScripts=true`? we should probably leave it to the user to decide whether or not they wanna get h4x0rd

Reply:
HTML blocks were originally a way to say, hey this code is dangerous and we know it. All other 'inlined' html would get tags and attributes sanitized. But I don't see how we could even begin to sanitize MDX.

The _only_ thing I can think of, is if some existing page with an html block has an unknown malicious script in it, and all of sudden we gave it life?